Feature/forum topics fetch

BytePaul opened the pull request:
This PR adds a new "Forum Browser" page under the Community section.  
1. Introduces /community-forum-browser.html with search, filter, and sort for mirrored Discourse topics  
2. Adds link to Community sidebar  
3. Includes tools/fetch_precice_forum.py to update assets/data/forum/topics.json

BytePaul committed:
Added a Page for the Accumulation of Discussions from Discourse

BytePaul committed:
Fixed issues in the forum-browser.md and added some comments

MakisH commented on _data/sidebars/community_sidebar.yaml:
```diff
@@ -119,6 +119,10 @@ entries:
     
+  - title: Forum Browser
+    url: /community-forum-browser.html
+    output: web
```
A more fitting place would be in the `docs_sidebar.yaml`, under `Fundamentals`, between `Roadmap` and `Previous versions`.

A better title would be `FAQ` and initialized with the `faq` tag in all categories.

BytePaul replied:
I have made the improvements as suggested. Let me know if it works.

BytePaul commented:
This is how the new addition "Forum Browser" looks like

<img width="1543" height="931" alt="precice_forum_page" src="https://github.com/user-attachments/assets/da84d69d-a167-4862-93d9-ff190d575bdb" />

MakisH commented on assets/data/forum/topics.json:
This data should not be directly in the repository. It should either:

a) be always loaded on demand when the page is loading, or
b) be stored in Git LFS and updated periodically with a GitHub Action.

Option (a) would be the easier to implement, option (b) could potentially be faster. Let's start with option (a).

MakisH replied:
The `check old API us` pre-commit hook is also failing, because it is picking up keywords in the JSON file. Removing the JSON file from the repository (and loading it every time it is needed) would fix that.

MakisH replied:
@BytePaul this file still needs to be removed from this branch (and the system needs to fetch the data on demand).

See https://github.com/precice/precice.github.io/pull/576 for an example of how to implement a button to fetch the data. Rendered on https://precice.org/fundamentals-roadmap.html

MakisH commented on css/customstyles-precice.css:
This originates from #582. Please rebase on `master` and remove the unrelated files.

MakisH replied:
@BytePaul FYI, I will wait for the cleanup before I continue with the rest of my review.

MakisH replied:
@BytePaul the changes to this file still need to be removed from this branch.

MakisH reviewed (commented):
Thanks for this PR, looks very nice!
Some first comments to consider. I will do a more thorough review later.

BytePaul committed:
docs: move Forum Browser to Fundamentals as FAQ

- Relocated sidebar entry from community_sidebar.yaml to docs_sidebar.yaml
- Placed FAQ under Fundamentals, between Roadmap and Previous versions
- Renamed 'Forum Browser' to 'FAQ'
- Added faq tag to enable categorization

MakisH committed:
Merge branch 'master' into feature/forum-topics-fetch

MakisH commented on pages/community/forum-browser.md:
This file is identical (duplicate) to `pages/docs/faq.html`. I would delete it.

MakisH commented on content/docs/fundamentals/fundamentals-faq.md:
```diff
@@ -0,0 +1,140 @@
+
+<script>
+  // Path to the generated JSON file
+  const dataUrl = "{{ '/assets/data/forum/topics.json' | relative_url }}";
```
The current state gives an error, as it still expects the file to be part of the repository (which we don't want):

```
ERROR `/assets/data/forum/topics.json' not found.
```

Instead, this file should be generated on demand (either by default, or when a user clicks a button).

MakisH commented on content/docs/fundamentals/fundamentals-faq.md:
Note that, recently, the `pages/` directory has been renamed to `content/`. We need to rename this directory as well.

MakisH committed:
Remove duplicate file

MakisH committed:
Move fundamentals-faq.md

MakisH committed:
Adjust FAQ page header

MakisH commented:
@BytePaul I adjusted a few details. The main missing part is the fetching of data (see thread above).

MakisH commented:
Superseded by #612.
